unity/CAVE/Assets/W3D_Translator/CLI.cs

```diff
@@ -207,55 +209,64 @@ private static void BuildWalls(Story xml, Transform storyT)
         /** Object
             name: gameObject.name
             Visible: gameObject.active
-            Color: gameObject.[content].color
+            Color: gameObject.[content].color (disabledColor if <LinkRoot> is present)
             Lighting: TODO (76)
             ClickThrough: TODO (76)
             AroundSelfAxis: TODO (76)
             Scale: gameObject.localScale (set in Placement.SetTransform)
         */
         foreach (W3D.Object xml in objectList)
         {
-            GameObject gameObject = new()
+            GameObject contentGO = xml.Content.Create(xml);
+
+            // TODO (74): LinkRoot.Link -> Add a VRCanvas
+            if (xml.LinkRoot is not null)
             {
-                name = xml.Name
-            };
-            xml.Placement.SetTransform(gameObject.transform, xml.Scale, story.transform);
+                // Instantiate a new link prefab
+                GameObject prefab = Instantiate(Resources.Load<GameObject>("Prefabs/canvas"));
+                prefab.GetComponent<Canvas>().worldCamera = UnityEngine.Camera.main;
+
+                // Set xml for canvas
+                prefab.name = xml.Name;
+                prefab.SetActive(xml.Visible);
+                xml.Placement.SetTransform(prefab.transform, xml.GetScale(), story.transform);
+                prefab.transform.localScale *= 0.1f;
+
+                Link link = xml.LinkRoot.Link;
+                GameObject buttonGO = prefab.transform.GetChild(0).gameObject;
+                Button button = buttonGO.GetComponent<Button>();
```

Comment:
You need a canvas to have a button? Is that how UIs work in Unity?

Reply:
Yeah, I think with a traditional UI you'd have 1 canvas with all of your buttons, sliders, input, etc laid out across that canvas. But because so many of our objects are moving around in 3D space I figured it would be best to just use 1 canvas for every element?